sdk/resourcemanager/Azure.ResourceManager.Core/docs/MigratingFromCurrentToPreview.md

```diff
@@ -0,0 +1,86 @@
+# Differences between current and preview .NET Management SDK
+
+## Structured Resource Identifier
+In the past, the resource identifier returned by Azure (often in a property named `Id`) was returned as a string. This made it difficult for users to retrieve specific pieces of the identifier, since they needed to implement their own parsing logic. Today, you can instead implicitly cast the string into an object, which will do the parsing for you. There are three types of ResourceIdentifiers, and they correspond to the level at which the resource lives. A resource that lives on a tenant will have a **TenantResourceIdentifier**. A resource that lives under a subscription will have a **SubscriptionResourceIdentifer**. A resource that lives under a resource group will have a **ResourceGroupResourceIdentifier**.
+
+You can usually tell by the `id` string itself which type it is. If you're unsure, you can cast it onto a **ResourceIdentifier** and use the `Try` methods to retrieve the values.
+
+### Casting to a specific type
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet";
+// We know the subnet is a resource group level identifier since it has a resource group name in its string
+ResourceGroupResourceIdentifier id = resourceId;
+Console.WriteLine($"Subscription: {id.SubscriptionId}");
+Console.WriteLine($"ResourceGroup: {id.ResourceGroupName}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
```

Comment:
weare not highlighting the fact that our actual methods and model types use these identifiers, and they are interchangeable with strings.